### PR TITLE
Add initial JenkinsFile for redux

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
In order to test I need get this merged :)

#### Ticket Link
jira: https://mattermost.atlassian.net/browse/MM-7495

### CODE REVIEW

File: build/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        docker exec mattermost-mysql-57 mysql --user=root --password=passwd -e 'create database `'$BUILD_TAG'`;'
+        """
+        echo 'Created database for mysql'
+        sh """
+        # Download and configure Mattermost
+        wget https://releases.mattermost.com/mattermost-platform/master/mattermost-enterprise-linux-amd64.tar.gz
+        tar -zxf mattermost-enterprise-linux-amd64.tar.gz
+
+        cd mattermost
+        sed -i'' 's|mmuser:mostest@tcp(dockerhost:3306)/mattermost_test?charset=utf8mb4,utf8|root:passwd@tcp(localhost:3306)/'${BUILD_TAG}'?charset=utf8mb4,utf8|g' config/config.json;
```
Comment (lieut-data) [inline build/Jenkinsfile]: I suspect this is from other such files, but I wonder if we should adopt a less brittle approach and just export the desired variables to the environment vs. messing with the config.json syntax:
```
export MM_SQLSETTINGS_DATASOURCE="..."
export MM_TEAMSETTINGS_ENABLEOPENSERVER="true"
...
```

Comment (cpanato) [reply build/Jenkinsfile]: agree

Comment (cpanato) [reply build/Jenkinsfile]: done and thanks so much for this valuable info!

**Comment by cpanato** [comment]
actually not! :) i will test and see any improvements/issues

**Comment by cpanato** [comment]
@lieut-data @jwilander PTAL, i've update based in the tests using the pipeline.

however, to test using the real server we need to fix some tests it is failing in the beginning

```Bash
> mattermost-redux@1.2.0 test-no-mock /home/ubuntu/jenkins/workspace/mmrdx/test-pipe-redux
> TEST_SERVER=1 NOCK_OFF=true NODE_ENV=test mocha --opts test/mocha.opts

/home/ubuntu/jenkins/workspace/mmrdx/test-pipe-redux/test/test_helper.js:197
                creator_id: _this.basicUser.id,
                                            ^

TypeError: Cannot read property 'id' of null
    at TestHelper.testCommand (/home/ubuntu/jenkins/workspace/mmrdx/test-pipe-redux/test/test_helper.js:194:40)
    at Suite.<anonymous> (/home/ubuntu/jenkins/workspace/mmrdx/test-pipe-redux/test/selectors/integrations.test.js:22:66)
    at Object.create (/home/ubuntu/jenkins/workspace/mmrdx/test-pipe-redux/node_modules/mocha/lib/interfaces/common.js:112:19)
    at context.describe.context.context (/home/ubuntu/jenkins/workspace/mmrdx/test-pipe-redux/node_modules/mocha/lib/interfaces/bdd.js:44:27)
    at Object.<anonymous> (/home/ubuntu/jenkins/workspace/mmrdx/test-pipe-redux/test/selectors/integrations.test.js:10:1)
    at Module._compile (module.js:652:30)
    at loader (/home/ubuntu/jenkins/workspace/mmrdx/test-pipe-redux/node_modules/babel-register/lib/node.js:144:5)
    at Object.require.extensions.(anonymous function) [as .js] (/home/ubuntu/jenkins/workspace/mmrdx/test-pipe-redux/node_modules/babel-register/lib/node.js:154:7)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at /home/ubuntu/jenkins/workspace/mmrdx/test-pipe-redux/node_modules/mocha/lib/mocha.js:253:27
    at Array.forEach (<anonymous>)
    at Mocha.loadFiles (/home/ubuntu/jenkins/workspace/mmrdx/test-pipe-redux/node_modules/mocha/lib/mocha.js:250:14)
    at Mocha.run (/home/ubuntu/jenkins/workspace/mmrdx/test-pipe-redux/node_modules/mocha/lib/mocha.js:577:10)
    at Object.<anonymous> (/home/ubuntu/jenkins/workspace/mmrdx/test-pipe-redux/node_modules/mocha/bin/_mocha:591:18)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:191:16)
    at bootstrap_node.js:612:3
npm ERR! code ELIFECYCLE
```

**Comment by cpanato** [comment]
ok, then I will comment the part that tests using a live server. ok @lieut-data ?

**Comment by lieut-data** [comment]
I'm personally ok with that! @jwilander, any objections?

**Comment by jwilander** [comment]
Yes that's good with me 👍

**Comment by cpanato** [comment]
@jwilander @lieut-data set only the master branch to run the tests with live server.

**Comment by jwilander** [comment]
@cpanato feel free to merge

**Pull request merged**